internal/runtime/executor/claude_executor_cloaking.go

```
@@ -431,6 +431,79 @@ func validateClaudeCallerSystemBlocks(system gjson.Result) error {
+	switch content.Get("type").String() {
+	case "server_tool_use", "advisor_tool_result", "web_search_tool_result",
+		"web_fetch_tool_result", "code_execution_tool_result",
+		"bash_code_execution_tool_result", "text_editor_code_execution_tool_result":
+		return true
+	}
+	return claudeContentContainsSignedServerState(content.Get("content"))
+}
+
+// claudeRequestContainsAdvisorState remains the recursive advisor detector used
+// by tests and callers that need to distinguish advisor from other server tools.
+func claudeRequestContainsAdvisorState(payload []byte) bool {
```

golangci-lint (2.13.2)

[error] func claudeRequestContainsAdvisorState is unused

(unused)